File: src/Entity/Depot.php
<?php
// src/Entity/Depot.php

namespace App\Entity;

use App\Entity\User;
use App\Repository\DepotRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: DepotRepository::class)]
#[ORM\Table(name: 'depots')]
#[ORM\HasLifecycleCallbacks]
class Depot
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $title = null;

    #[ORM\Column(type: 'text', nullable: true)]
    #[Assert\Length(
        max: 500,
        maxMessage: "La description ne peut pas dépasser 500 caractères"
    )]
    private ?string $description = null;

    #[ORM\Column(length: 50)]
    #[Assert\Choice(
        choices: ['document', 'code', 'presentation', 'rapport', 'autre'],
        message: "Le type doit être: document, code, presentation, rapport ou autre"
    )]
    private ?string $type = 'document';

    #[ORM\Column(length: 255)]
    private ?string $filePath = null;

    #[ORM\Column(length: 50, nullable: true)]
    private ?string $fileSize = null;

    #[ORM\Column(length: 100, nullable: true)]
    private ?string $fileType = null;

    #[ORM\Column(type: 'datetime')]
    private ?\DateTimeInterface $uploadedAt = null;

    #[ORM\ManyToOne(targetEntity: Project::class, inversedBy: 'depots')]
    #[ORM\JoinColumn(nullable: false, onDelete: 'CASCADE')]
    private ?Project $project = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id', nullable: true, onDelete: 'SET NULL')]
    private ?User $user = null;

    #[ORM\Column(length: 100)]
    private ?string $studentName = null;

    #[ORM\Column(type: 'integer', options: ['default' => 0])]
    private int $downloadCount = 0;

    #[ORM\PrePersist]
    public function setUploadedAtValue(): void
    {
        $this->uploadedAt = new \DateTime();
    }

    // Getters et Setters
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(string $title): static
    {
        $this->title = $title;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;
        return $this;
    }

    public function getFilePath(): ?string
    {
        return $this->filePath;
    }

    public function setFilePath(string $filePath): static
    {
        $this->filePath = $filePath;
        return $this;
    }

    public function getFileSize(): ?string
    {
        return $this->fileSize;
    }

    public function setFileSize(?string $fileSize): static
    {
        $this->fileSize = $fileSize;
        return $this;
    }

    public function getFileType(): ?string
    {
        return $this->fileType;
    }

    public function setFileType(?string $fileType): static
    {
        $this->fileType = $fileType;
        return $this;
    }

    public function getUploadedAt(): ?\DateTimeInterface
    {
        return $this->uploadedAt;
    }

    public function setUploadedAt(\DateTimeInterface $uploadedAt): static
    {
        $this->uploadedAt = $uploadedAt;
        return $this;
    }

    public function getProject(): ?Project
    {
        return $this->project;
    }

    public function setProject(?Project $project): static
    {
        $this->project = $project;
        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): static
    {
        $this->user = $user;
        return $this;
    }

    public function getStudentName(): ?string
    {
        return $this->studentName;
    }

    public function setStudentName(string $studentName): static
    {
        $this->studentName = $studentName;
        return $this;
    }

    public function getDownloadCount(): int
    {
        return $this->downloadCount;
    }

    public function setDownloadCount(int $downloadCount): static
    {
        $this->downloadCount = $downloadCount;
        return $this;
    }

    public function incrementDownloadCount(): static
    {
        $this->downloadCount++;
        return $this;
    }

    // Méthodes utilitaires
    public function getTypeIcon(): string
    {
        return match ($this->type) {
            'document' => 'fa-file-pdf',
            'code' => 'fa-code',
            'presentation' => 'fa-file-powerpoint',
            'rapport' => 'fa-file-alt',
            default => 'fa-file'
        };
    }

    public function getTypeColor(): string
    {
        return match ($this->type) {
            'document' => 'text-red-600',
            'code' => 'text-blue-600',
            'presentation' => 'text-orange-600',
            'rapport' => 'text-green-600',
            default => 'text-gray-600'
        };
    }

    public function getFormattedFileSize(): string
    {
        if (!$this->fileSize)
            return 'Inconnu';

        $bytes = (int) $this->fileSize;
        if ($bytes >= 1073741824) {
            return number_format($bytes / 1073741824, 2) . ' GB';
        } elseif ($bytes >= 1048576) {
            return number_format($bytes / 1048576, 2) . ' MB';
        } elseif ($bytes >= 1024) {
            return number_format($bytes / 1024, 2) . ' KB';
        } else {
            return $bytes . ' bytes';
        }
    }
}

File: src/main/java/com/innolearn/model/Depot.java
package com.innolearn.model;

import java.sql.Timestamp;

public class Depot {
    private int id;
    private String title;
    private String description;
    private String type;
    private String filePath;   // From SQL: file_path
    private String fileSize;   // From SQL: file_size
    private String fileType;   // From SQL: file_type
    private Timestamp uploadedAt; // From SQL: uploaded_at
    private int projectId;     // From SQL: project_id
    private String studentName; // From SQL: student_name
    private int downloadCount;  // From SQL: download_count
    private Integer userId;     // From SQL: user_id (nullable)
    private String todoStatus;   // From SQL: todo_status (Done | Doing | Didn't Do)
    private String aiResult;     // From SQL: ai_result (full AI feedback text)
    private Integer aiScore;     // From SQL: ai_score (0-100)

    public Depot() {}

    public Depot(int id, String title, String description, String type, String filePath, String fileSize, String fileType, Timestamp uploadedAt, int projectId, String studentName, int downloadCount, Integer userId) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.type = type;
        this.filePath = filePath;
        this.fileSize = fileSize;
        this.fileType = fileType;
        this.uploadedAt = uploadedAt;
        this.projectId = projectId;
        this.studentName = studentName;
        this.downloadCount = downloadCount;
        this.userId = userId;
        this.todoStatus = "Doing";
    }

    public Depot(int id, String title, String description, String type, String filePath, String fileSize, String fileType, Timestamp uploadedAt, int projectId, String studentName, int downloadCount, Integer userId, String todoStatus) {
        this(id, title, description, type, filePath, fileSize, fileType, uploadedAt, projectId, studentName, downloadCount, userId);
        this.todoStatus = todoStatus;
    }

    // Getters and Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getFilePath() { return filePath; }
    public void setFilePath(String filePath) { this.filePath = filePath; }
    public String getFileSize() { return fileSize; }
    public void setFileSize(String fileSize) { this.fileSize = fileSize; }
    public String getFileType() { return fileType; }
    public void setFileType(String fileType) { this.fileType = fileType; }
    public Timestamp getUploadedAt() { return uploadedAt; }
    public void setUploadedAt(Timestamp uploadedAt) { this.uploadedAt = uploadedAt; }
    public int getProjectId() { return projectId; }
    public void setProjectId(int projectId) { this.projectId = projectId; }
    public String getStudentName() { return studentName; }
    public void setStudentName(String studentName) { this.studentName = studentName; }
    public int getDownloadCount() { return downloadCount; }
    public void setDownloadCount(int downloadCount) { this.downloadCount = downloadCount; }
    public Integer getUserId() { return userId; }
    public void setUserId(Integer userId) { this.userId = userId; }
    public String getTodoStatus() { return todoStatus; }
    public void setTodoStatus(String todoStatus) { this.todoStatus = todoStatus; }
    public String getAiResult() { return aiResult; }
    public void setAiResult(String aiResult) { this.aiResult = aiResult; }
    public Integer getAiScore() { return aiScore; }
    public void setAiScore(Integer aiScore) { this.aiScore = aiScore; }

    @Override
    public String toString() {
        return "Depot{" + "id=" + id + ", title='" + title + '\'' + ", projectId=" + projectId + '}';
    }
}
